### PR TITLE
Cleans up + Improves bows, Sorts files, Adds the Divine Archer clothing, weapon, rite

### PR DESCRIPTION
**Pull request by tralezab**
### Divine Archer 🏹 

image

Adds a new chaplain weapon and suit of armor, the divine archer. It's an orderable set of armor, but provides less armor than the rest, but you get more pieces of armor (boots, bracer, undersuit).

The divine bow comes with a quiver that holds holy arrows. The holy arrows come with bane support, dealing critical damage to revenants.

### Bow Features ⭐ 

- arrows can now be dipped in poison

### Bow Improvements 🔧 

- bows now drop their arrow when you put them on your back while nocking a bow
- bows give feedback for trying to draw without a nocked arrow
- codewise, bows support subtypes much better. They still have hard-sprited loaded arrows, but one day that'll change.

## Why It's Good For The Game

Yeah, we could add null rod #2342 that does almost the same as the others, or we could have a unique bow weapon!

Player Dev Project thread: https://discord.com/channels/326822144233439242/1093521091957370940/1093521091957370940

## Changelog
:cl: tralezab code, Drag for the commission and player project, cre#0484 for their spritework
add: Divine Archer Armor and Weapon
qol: Bows give more feedback when you're doing something wrong, like trying to draw without a nocked arrow
code: Sorted files, cleaned bow code up to allow subtypes
/:cl:

### CODE REVIEW

File: code/modules/jobs/job_types/chaplain/chaplain_nullrod.dm [hunk 1]
```diff
@@ -39,13 +39,16 @@
...
+		//special non-nullrod subtyped shit
+		rods[/obj/item/gun/ballistic/bow/divine/with_quiver] = "A divine bow and 10 quivered holy arrows."
 		AddComponent(/datum/component/subtype_picker, rods, CALLBACK(src, PROC_REF(on_holy_weapon_picked)))
 
 /obj/item/nullrod/proc/on_holy_weapon_picked(obj/item/nullrod/holy_weapon_type)
 	GLOB.holy_weapon_type = holy_weapon_type
 	SSblackbox.record_feedback("tally", "chaplain_weapon", 1, "[initial(holy_weapon_type.name)]")
 
 /obj/item/nullrod/proc/on_cult_rune_removed(obj/effect/target, mob/living/user)
+	SIGNAL_HANDLER
```
[the remark this answers is not in the document]
Comment (tralezab) [reply code/modules/jobs/job_types/chaplain/chaplain_nullrod.dm:51]: on_ brain auto placing SIGNAL HANDER

**Comment by Draggeru** [comment]
Capitalize my name in the credits, coward (Joke)

**Comment by tralezab** [comment]
done 👍

**Comment by necromanceranne** [comment]
I didn't even know there wasn't a projectile bane proc. Nice dude.

Also, is there a way to recover arrows or get more arrows, or is it just the roundstart 10? Would it be unreasonable to add a way to get more of them?

**Comment by tralezab** [comment]
> Crafting

I might expand upon this in the future with rudimentary and non-rudimentary arrows, but it won't be strictly chaplain related

> Cargo

I don't think it really fits cargo's deal, but I could see a world where getting the weapon as chappy unlocks a crate in cargo to buy more. My only worry is that I'd 100% have to nerf the quiver, it currently holds an insane amount of arrows and the previous ideology was that arrows don't come back

> Sect

Interesting! Sadly, I can't add a generic rite because sects have unique mechanics and prices, sometimes offering no way to get favor at all for a completely different mechanic. But playing with the idea:
- puritanism is meant to be locking in as "no sect", so nothing special
- mechanical... really doesn't fit.
- pyre... i could see a cool new rite to buff arrows, but not to make new ones.
- greed? yeah, they could become purchasable.
- punished? really too hard to implement, and finishing the sect means you're too blind to use a bow anyways. and you have a gun. there's a lot of reasons why it's not gonna work
- honorbound? I really like it, it fits thematically, the only issue is that they require fair fights. Bows... make that a little messy. I mean, I really might still do it, but I predict some not entirely honorbound behavior from it.
- maintenance? I like it, but ghetto arrows, not holy ones. They should work with the holy bow but I need to refactor how bows handle the arrow sprites for it.
- sparring god? Could be a sparring reward. Not a bad idea. Probably the most viable
- festival god? very, very unrelated. I mean maybe. How would that work? I don't want a song that summons ammo, I don't want rites that aren't song tuners, there's really not a lot of budging here. Maybe... a gun reload song? That just sounds stupid, but that's as close as I'll get

but overall, it seems like a noob trap to tie arrow renewal to *some* sects. I could look into ways to get normal arrows and bless them, but even that is... eh. Then it's too easy to get new arrows. Maybe I should just make the holy arrows recallable after a very long channel or something, or give them a very large stock.

**Comment by Wallemations** [comment]
Damn, I was gonna remake bows but you beat me to it
I wanted to make them like the TF2 huntsman where you can charge a shot for more damage

**Comment by tralezab** [comment]
> Nobody has mentioned that your link to the other bow PR is definitely not the right PR.

oops, lol, that wasn't meant to be a PR link. just saying that we add a billion uninteresting chappy weapons, thats all

**Comment by Elli-Skala** [comment]
Something more simple would be just using a Bible on a arrow makes it holy (like holy water)

**Comment by Elli-Skala** [comment]
Actually thinking about this, I think it might be to weak while most null rods are in the ~18 (6 hits) damage area this is 20 with six times the amount of clicks(swap hands, grab arrow, notch arrow, swap hands, pull arrow, fire arrow. Five hits not counting armor) , a naked assistant punching you would likely end you first even if you're making perfect hand swaps and arrow notches. There is a reason why a normal bow is a whopping 50 damage

**Comment by tralezab** [comment]
Sure, but it's a ranged weapon, so it needs to perform slightly less than melee. There's poison dipping, and pyre blessed arrows for some pursuits to increase deadliness as well

**Review by Imaginos16 — APPROVED** [review]
Cre's spritework like always, is absolutely gorgeous!

File: code/modules/religion/pyre_rites.dm [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+		"... make it burn bright ...",
+	)
+	invoke_msg = "... a blazing star is born!"
+	favor_cost = 2000
+	///arrow to enchant
+	var/obj/item/ammo_casing/caseless/arrow/holy/enchant_target
+
+/datum/religion_rites/blazing_star/perform_rite(mob/living/user, atom/religious_tool)
+	for(var/obj/item/ammo_casing/caseless/arrow/holy/can_enchant in get_turf(religious_tool))
+		if(istype(can_enchant, /obj/item/ammo_casing/caseless/arrow/holy/blazing))
```
Comment (Watermelon914) [inline code/modules/religion/pyre_rites.dm:132]: hmm? does this check need to exist? can't you just do the typecheck in the line above this

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

**Comment by celulamp** [comment]
Been using this it's awesome.
Would like to see some of the other kits get special weapons too.

**Pull request merged**